src/Cake.Core/CakeEnvironment.cs

```diff
@@ -104,23 +100,7 @@ public string GetEnvironmentVariable(string variable)
                     new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase),
                     (dictionary, entry) =>
                     {
-                        var key = (string)entry.Key;
-                        var value = entry.Value as string;
-                        if (dictionary.TryGetValue(key, out var existingValue))
-                        {
-                            if (!StringComparer.OrdinalIgnoreCase.Equals(value, existingValue))
-                            {
-                                _log.Warning("GetEnvironmentVariables() encountered duplicate for key: {0}, value: {1} (existing value: {2})",
-                                    key,
-                                    value,
-                                    existingValue);
-                            }
-                        }
-                        else
-                        {
-                            dictionary.Add(key, value);
-                        }
```

Comment:
Not possible to have duplicate environment variables, so removed this code. Unnecessary to require a log instance for the environment as well since it's one of those classes that shouldn't require anything that in theory might require itself.

Reply:
It is possible as environment variables is just a string in some OS:s i.e. we had issue cake wouldn't start on systems with "corrupt" variables, caused by some APIs had different case sensitivity. So you could get duplicate keys between system/user/process env variables.

Reply:
Ok, I need to think about this a bit. Can't require environment for the logger when we require the logger for the environment.

Reply:
I think this is a bit of an edge case perhaps just `return dictionary;` if found? And skip the logging.

Reply:
@devlead Can you elaborate what you mean with ”return the dictionary if found”?